test/ListUtils.java

```diff
@@ -0,0 +1,32 @@
+package test;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public final class ListUtils {
+
+    private ListUtils() {
+    }
+
+    public static <T> void swapElemWithImmutable(List<T> source, List<T> origin, int index) {
```

Comment:
>Create utility method, which takes two lists and index and replaces element at given position in the first collection with corresponding element from the second one, without possibility to do vice versa;

1) Take element from second collection and set it to the first collection.
2) The goal is not to create unmodifiable list but to forbid writing such code:
```
origin.set(index, source.get(index));
```
It shouldn't compile.